### PR TITLE
Add arm32 tp

### PR DESCRIPTION
**Pull request by adiaaida**

### CODE REVIEW

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

File: perf.groovy [hunk 1]
```diff
@@ -396,169 +403,224 @@ parallel(
...
-                shell("./build.sh verbose ${architecture} ${configuration}")
+        
+        def crossCompile = ""
+        def python = "python3.5"
+
+        if (architecture == "arm") {
+            python = "python3.6"
+            def buildCommands = []
+            def newBuildJob = job(fullBuildJobName) {
+                // arm build is run on docker machine, so machine affinity doesn't matter
```
Comment (jashook) [inline perf.groovy]: Seems like the comment isn't necessary, the machine affinity is being set https://github.com/dotnet/coreclr/pull/19098/files#diff-bc9a87b2e014aa8266aab4a55fbbc50cR473

File: perf.groovy [hunk 2]
```diff
@@ -396,169 +403,224 @@ parallel(
...
-                        buildSelector {
-                            buildNumber('\${PRODUCT_BUILD}')
+                    def osGroup = getOSGroup(os)
+                    def runType = isPR ? 'private' : 'rolling'
+                    def benchViewName = isPR ? 'coreclr-throughput private \$BenchviewCommitName' : 'coreclr-throughput rolling \$GIT_BRANCH_WITHOUT_ORIGIN \$GIT_COMMIT'
+                    def archString = architecture == 'arm' ? ' --arch=arm' : ''
+
+                    steps {
+                        shell("bash ./tests/scripts/perf-prep.sh --throughput${archString}")
+                        //shell("./init-tools.sh")
```
Comment (jashook) [inline perf.groovy]: Why is this commented?

File: tests/scripts/run-throughput-perf.py [hunk 3]
```diff
@@ -166,12 +173,12 @@ def validate_arg(arg, check):
...
             raise Exception('Argument: %s is not valid.' % (arg))
 
-    valid_archs = {'Windows_NT': ['x86', 'x64'], 'Linux': ['x64']}
+    valid_archs = {'Windows_NT': ['x86', 'x64'], 'Linux': ['x64', 'arm']}
     valid_build_types = ['Release']
     valid_run_types = ['rolling', 'private']
     valid_os = ['Windows_NT', 'Ubuntu14.04', 'Ubuntu16.04']
     valid_opt_levels = ['full_opt', 'min_opt']
-    valid_jit_names = {'x64': ['ryujit'], 'x86': ['ryujit', 'legacy_backend']}
+    valid_jit_names = {'x64': ['ryujit'], 'x86': ['ryujit', 'legacy_backend'], 'arm': ['ryujit', 'legacy_backend']}
```
Comment (jashook) [inline tests/scripts/run-throughput-perf.py]: I believe legacy_backend can safely be removed @BruceForstall to confirm

File: tests/scripts/run-throughput-perf.py [hunk 4]
```diff
@@ -300,9 +307,9 @@ def runIterations(dll_name, dll_path, iterations, crossgen_path, jit_path, assem
     if opt_level == 'min_opt':
         my_env['COMPlus_JITMinOpts'] = '1'
 
-    if jit_name == 'legacy_backend':
-        my_env['COMPlus_AltJit'] = '*'
-        my_env['COMPlus_AltJitNgen'] = '*'
+    #if jit_name == 'legacy_backend':
```
Comment (jashook) [inline tests/scripts/run-throughput-perf.py]: Probably should just be removed.

**Comment by jashook** [comment]
lgtm minus a few nits

File: perf.groovy (hunk 1 above)
Comment (adiaaida) [reply perf.groovy]: I just copied that from netci.groovy. Will remove

File: perf.groovy (hunk 2 above)
Comment (adiaaida) [reply perf.groovy]: It doesn't run properly on arm. And doesn't seem to be necessary for throughput. Will test on x64 tp to make sure that it isn't needed at all then remove the line altogether

**Comment by adiaaida** [comment]
@dotnet-bot test linux throughput flow please

File: tests/scripts/run-throughput-perf.py (hunk 3 above)
Comment (jorive) [reply tests/scripts/run-throughput-perf.py]: I believe all references to `legacy_backend` can be safely removed.

Comment (adiaaida) [reply tests/scripts/run-throughput-perf.py]: Cool. Removed.

**Comment by BruceForstall** [comment]
Not sure I can help with the review, but I can say: thank you for getting this going! Looking forward to arm32 (and soon arm64?) getting first-class support built out.

**Pull request merged**